include/RAJA/exec-cuda/MemUtils_CUDA.hxx

```diff
@@ -62,37 +62,90 @@
 
 namespace RAJA
 {
+/// Maximum number of blocks that RAJA will launch
+/// threads are reused to finish all work
+#define RAJA_CUDA_MAX_NUM_BLOCKS (1024 * 16)
 
 /// Size of reduction memory block for each reducer object (value based on
 /// rough estimate of "worst case" -- need to think more about this...
-#define RAJA_CUDA_REDUCE_BLOCK_LENGTH (1024 + 8) * 16
+#define RAJA_CUDA_REDUCE_BLOCK_LENGTH RAJA_CUDA_MAX_NUM_BLOCKS
 
 /// Reduction Tallies are computed into a small block to minimize UM migration
 #define RAJA_CUDA_REDUCE_TALLY_LENGTH RAJA_MAX_REDUCE_VARS
 
-///
-/// Typedef defining common data type for RAJA-Cuda reduction data blocks
-/// (use this in all cases to avoid type confusion).
-///
-typedef double CudaReductionBlockDataType;
+/// Should be large enough for all types for which cuda atomics exist
+/// includes the size of the index variable for Loc reductions
+#define RAJA_CUDA_REDUCE_VAR_MAXSIZE 16
+
```

Comment:
Please add documentation comments (using Doygen delimiters -- we eventually want to generate this documentation) to all types, #defines, typedefs, etc. where they are missing.

Part of our issue discussion at the last meeting centered around documentation. We may as well start addressing this